### PR TITLE
Migrate to TypeScript

### PR DESCRIPTION
**Pull request by AbdSat**
Migrate this project to TypeScript and provide type declarations.

**Resolving [#82](https://github.com/editor-js/paragraph/issues/82)**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+  /**
+   * Whether or not to keep blank paragraphs when saving editor data
+   */
+  private readonly preserveBlank: boolean;
+
+  /**
+   * Default placeholder for Paragraph Tool
+   *
+   * @class
```
Comment (TatianaFomina) [inline src/index.ts:66]: seems like `@class` is not needed

**Review by TatianaFomina — CHANGES_REQUESTED** [review]
Tried to build and import this version of the plugin, seems like type declarations are still  missing... 

image

File: paragraph.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+declare module '@editorjs/paragraph';
```
Comment (TatianaFomina) [inline paragraph.d.ts:1]: This file should be generated during build time and should contain all the types. Investigate on ways to do this, maybe use [this vite plugin ](https://github.com/qmhc/vite-plugin-dts)

**Review by TatianaFomina — CHANGES_REQUESTED** [review]

File: vite.config.js [hunk 3]
```diff
@@ -9,17 +10,20 @@ export default {
...
     copyPublicDir: false,
     lib: {
-      entry: path.resolve(__dirname, 'src', 'index.js'),
+      entry: path.resolve(__dirname, 'src', 'index.ts'),
       name: 'Paragraph',
       fileName: 'paragraph',
+      formats: ['umd', 'es'],
     },
   },
+
```
Comment (TatianaFomina) [inline vite.config.js:19]:
```suggestion
```

**Review by TatianaFomina — CHANGES_REQUESTED** [review]
Doesn't seem to work 🥲

image

**Comment by neSpecc** [comment]
Code Review was not passed

**Pull request closed without merging**